### ISSUE TITLE
Missing `az connectedmachine run-command`

### ISSUE THREAD

**Issue by jkroepke** [Azure/azure-cli-extensions#8151]
### Describe the bug

After upgrading connectedmachine extension from 0.7.0 to 1.0.0, the commmand `az connectedmachine run-command` is missing.

### Related command

az connectedmachine run-command

### Errors

'run-command' is misspelled or not recognized by the system.

### Issue script & Debug output

DEBUG: cli.knack.cli: Command arguments: ['connectedmachine', 'run-command', '--help', '--debug']
DEBUG: cli.knack.cli: __init__ debug log:
Cannot enable color.
DEBUG: cli.knack.cli: Event: Cli.PreExecute []
DEBUG: cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x101610400>, <function OutputProducer.on_global_arguments at 0x1017b2840>, <function CLIQuery.on_global_arguments at 0x1017f0360>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
DEBUG: cli.azure.cli.core: Modules found from index for 'connectedmachine': ['azext_connectedmachine']
DEBUG: cli.azure.cli.core: Loading command modules:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
DEBUG: cli.azure.cli.core: Total (0)                 0.000         0         0
DEBUG: cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
DEBUG: cli.azure.cli.core: Loading extensions:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
DEBUG: cli.azure.cli.core: connectedmachine          0.116         6        13  /Users/jok/.azure/cliextensions/connectedmachine
DEBUG: cli.azure.cli.core: Total (1)                 0.116         6        13  
DEBUG: cli.azure.cli.core: Loaded 6 groups, 13 commands.
DEBUG: cli.azure.cli.core: Could not find a match in the command or command group table for 'connectedmachine run-command'. The index may be outdated.
DEBUG: cli.azure.cli.core: Loading all modules and extensions
DEBUG: cli.azure.cli.core: Discovered command modules: ['acr', 'acs', 'advisor', 'ams', 'apim', 'appconfig', 'appservice', 'aro', 'backup', 'batch', 'batchai', 'billing', 'botservice', 'cdn', 'cloud', 'cognitiveservices', 'compute_recommender', 'config', 'configure', 'consumption', 'container', 'containerapp', 'cosmosdb', 'databoxedge', 'dla', 'dls', 'dms', 'eventgrid', 'eventhubs', 'extension', 'feedback', 'find', 'hdinsight', 'identity', 'interactive', 'iot', 'keyvault', 'kusto', 'lab', 'managedservices', 'maps', 'marketplaceordering', 'monitor', 'mysql', 'netappfiles', 'network', 'policyinsights', 'privatedns', 'profile', 'rdbms', 'redis', 'relay', 'resource', 'role', 'search', 'security', 'servicebus', 'serviceconnector', 'servicefabric', 'signalr', 'sql', 'sqlvm', 'storage', 'synapse', 'util', 'vm']
DEBUG: cli.azure.cli.core: Loading command modules:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
DEBUG: cli.azure.cli.core: acr                       0.008        36       149
DEBUG: cli.azure.cli.core: acs                       0.014        14        77
DEBUG: cli.azure.cli.core: advisor                   0.001         3         6
DEBUG: cli.azure.cli.core: ams                       0.003        22       100
DEBUG: cli.azure.cli.core: apim                      0.003        14        69
DEBUG: cli.azure.cli.core: appconfig                 0.002         9        47
DEBUG: cli.azure.cli.core: appservice                0.047        79       270
DEBUG: cli.azure.cli.core: aro                       0.009         1        10
DEBUG: cli.azure.cli.core: backup                    0.002        16        60
DEBUG: cli.azure.cli.core: batch                     0.015        34       102
DEBUG: cli.azure.cli.core: batchai                   0.002        10        30
DEBUG: cli.azure.cli.core: billing                   0.010        20        53
DEBUG: cli.azure.cli.core: botservice                0.002        12        42
DEBUG: cli.azure.cli.core: cdn                       0.073         8        50
DEBUG: cli.azure.cli.core: cloud                     0.001         1         7
DEBUG: cli.azure.cli.core: cognitiveservices         0.001        10        33
DEBUG: cli.azure.cli.core: compute_recommender       0.003         1         1
DEBUG: cli.azure.cli.core: config                    0.002         2         7
DEBUG: cli.azure.cli.core: configure                 0.001         2         5
DEBUG: cli.azure.cli.core: consumption               0.013         8         9
DEBUG: cli.azure.cli.core: container                 0.007         1        11
DEBUG: cli.azure.cli.core: containerapp              0.066        37       123
DEBUG: cli.azure.cli.core: cosmosdb                  0.008        58       199
DEBUG: cli.azure.cli.core: databoxedge               0.007         5        28
DEBUG: cli.azure.cli.core: dla                       0.002        23        62
DEBUG: cli.azure.cli.core: dls                       0.002         7        41
DEBUG: cli.azure.cli.core: dms                       0.001         3        22
DEBUG: cli.azure.cli.core: eventgrid                 0.002        25        96
DEBUG: cli.azure.cli.core: eventhubs                 0.011        13        19
DEBUG: cli.azure.cli.core: extension                 0.001         1         7
DEBUG: cli.azure.cli.core: feedback                  0.001         1         2
DEBUG: cli.azure.cli.core: find                      0.001         1         1
DEBUG: cli.azure.cli.core: hdinsight                 0.005         9        42
DEBUG: cli.azure.cli.core: identity                  0.001         2        11
DEBUG: cli.azure.cli.core: interactive               0.000         1         1
DEBUG: cli.azure.cli.core: iot                       0.068        19        82
DEBUG: cli.azure.cli.core: keyvault                  0.005        20       113
DEBUG: cli.azure.cli.core: kusto                     0.002         3        14
DEBUG: cli.azure.cli.core: lab                       0.021         9        29
DEBUG: cli.azure.cli.core: managedservices           0.008         3         8
DEBUG: cli.azure.cli.core: maps                      0.001         5        13
DEBUG: cli.azure.cli.core: marketplaceordering       0.002         1         2
DEBUG: cli.azure.cli.core: monitor                   0.304        17        61
DEBUG: cli.azure.cli.core: mysql                     0.060        16        56
DEBUG: cli.azure.cli.core: netappfiles               0.027         8        17
DEBUG: cli.azure.cli.core: network                   0.111       103       338
DEBUG: cli.azure.cli.core: policyinsights            0.009         9        17
DEBUG: cli.azure.cli.core: privatedns                0.014        14        60
DEBUG: cli.azure.cli.core: profile                   0.001         2         8
DEBUG: cli.azure.cli.core: rdbms                     0.014        49       202
DEBUG: cli.azure.cli.core: redis                     0.002         7        38
DEBUG: cli.azure.cli.core: relay                     0.017         7         8
DEBUG: cli.azure.cli.core: resource                  0.007        51       231
DEBUG: cli.azure.cli.core: role                      0.002        17        61
DEBUG: cli.azure.cli.core: search                    0.006         7        19
DEBUG: cli.azure.cli.core: security                  0.007        48        98
DEBUG: cli.azure.cli.core: servicebus                0.006        12        14
DEBUG: cli.azure.cli.core: serviceconnector          0.018        20       309
DEBUG: cli.azure.cli.core: servicefabric             0.010        27        80
DEBUG: cli.azure.cli.core: signalr                   0.002         9        34
DEBUG: cli.azure.cli.core: sql                       0.009        56       215
DEBUG: cli.azure.cli.core: sqlvm                     0.019         4        20
DEBUG: cli.azure.cli.core: storage                   0.022        59       273
DEBUG: cli.azure.cli.core: synapse                   0.007        54       246
DEBUG: cli.azure.cli.core: util                      0.001         3         7
DEBUG: cli.azure.cli.core: vm                        0.038        58       233
DEBUG: cli.azure.cli.core: Total (66)                1.148      1206      4698
DEBUG: cli.azure.cli.core: Loading extensions:
DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
DEBUG: cli.azure.cli.core: connectedmachine          0.002         6        13  /Users/jok/.azure/cliextensions/connectedmachine
DEBUG: cli.azure.cli.core: Total (1)                 0.002         6        13  
DEBUG: cli.azure.cli.core: Loaded 1198 groups, 4711 commands.
DEBUG: cli.azure.cli.core: Updated command index in 0.004 seconds.
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x1024e2520>]
DEBUG: cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to '/Users/jok/.azure/commands/2024-10-16.12-23-36.connectedmachine.73730.log'.
INFO: az_command_data_logger: command args: connectedmachine {} --help --debug
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x1025300e0>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x1025336a0>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x1025337e0>, <function register_upcoming_breaking_change_info.<locals>.update_breaking_change_info at 0x102533880>]
DEBUG: cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreParseArgs [<function _documentdb_deprecate at 0x1040563e0>]
DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): app.aladdin.microsoft.com:443
DEBUG: cli.azure.cli.core.command_recommender: Recommendation requests.get() exception: HTTPSConnectionPool(host='app.aladdin.microsoft.com', port=443): Max retries exceeded with url: /api/v1.0/suggestions?query=%7B%22command%22%3A+%22connectedmachine+run-command%22%2C+%22parameters%22%3A+%22--help%22%7D&clientType=AzureCli&context=%7B%22versionNumber%22%3A+%222.65.0%22%2C+%22errorType%22%3A+%22UnknownSubcommand%22%2C+%22correlationId%22%3A+%22d869bd8d-b186-46df-9966-8a53fd68841c%22%2C+%22subscriptionId%22%3A+%22dde0e2ca-8b60-4211-85de-0218a399d6fe%22%2C+%22eventId%22%3A+%226fbda68b-7ae2-4409-ab27-9db651ea9e9b%22%7D (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x107d11b50>: Failed to establish a new connection: [Errno 8] nodename nor servname provided, or not known'))
DEBUG: cli.azure.cli.core.azclierror: NoneType: None

ERROR: cli.azure.cli.core.azclierror: 'run-command' is misspelled or not recognized by the system.
ERROR: az_command_data_logger: 'run-command' is misspelled or not recognized by the system.

Examples from AI knowledge base:
https://aka.ms/cli_ref
Read more about the command in reference docs
DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x1024e27a0>]
INFO: az_command_data_logger: exit code: 2
INFO: cli.__main__: Command ran in 1.432 seconds (init: 0.115, invoke: 1.317)
INFO: telemetry.main: Begin splitting cli events and extra events, total events: 1
INFO: telemetry.client: Accumulated 0 events. Flush the clients.
INFO: telemetry.main: Finish splitting cli events and extra events, cli events: 1
INFO: telemetry.save: Save telemetry record of length 3934 in cache file under /Users/jok/.azure/telemetry/20241016122336067
INFO: telemetry.main: Begin creating telemetry upload process.
INFO: telemetry.process: Creating upload process: "/opt/homebrew/Cellar/azure-cli/2.65.0/libexec/bin/python /opt/homebrew/Cellar/azure-cli/2.65.0/libexec/lib/python3.11/site-packages/azure/cli/telemetry/__init__.py /Users/jok/.azure /Users/jok/.azure/telemetry/20241016122336067"
INFO: telemetry.process: Return from creating process 73736
INFO: telemetry.main: Finish creating telemetry upload process.

### Expected behavior

It should work like on 0.7.0

### Environment Summary

azure-cli                         2.65.0

core                              2.65.0
telemetry                          1.1.0

Extensions:
connectedmachine                   1.0.0

Dependencies:
msal                              1.31.0
azure-mgmt-resource               23.1.1

Python location '/opt/homebrew/Cellar/azure-cli/2.65.0/libexec/bin/python'
Extensions directory '/Users/jok/.azure/cliextensions'

Python (Darwin) 3.11.10 (main, Sep  7 2024, 01:03:31) [Clang 15.0.0 (clang-1500.3.9.4)]

Legal docs and information: aka.ms/AzureCliLegal

### Additional context

Removed in https://github.com/Azure/azure-cli-extensions/pull/8026

**Comment by jkroepke** [comment]
The command

**Comment by Sorensiim** [comment]
We're having the exact same issue and the New-AzConnectedMachineRunCommand cmdlet was dropped from the PowerShell module 4 days ago when it went from 0.9.0 to 0.10.0. And with no way to install the 0.9.0 version in our runtime environment, it seems that we're up a certain creek experiencing a momentary shortage of paddles.

Maybe they discovered a security issue and decided to quickly just yank the cmdlets?

**Comment by jkroepke** [comment]
> Maybe they discovered a security issue and decided to quickly just yank the cmdlets?

@Sorensiim Based on the comment on https://github.com/Azure/azure-cli-extensions/pull/8026#issuecomment-2417387546, they just dropped all preview features.

**Comment by sassdawe** [comment]
> TL;DR: For AZ CLI

```
az extension remove --name connectedmachine
az extension add --name connectedmachine --version 0.7.0
```

resulting in `Version '0.7.0' not found for extension 'connectedmachine'`

![Image](https://github.com/user-attachments/assets/eafd7eb9-76ee-420a-86c6-1c2e9c9cbe1f)

Perfecto.
